source/MaterialXContrib/MaterialXNode/CreateMaterialXNodeCmd.cpp

```diff
@@ -116,78 +116,94 @@ MStatus CreateMaterialXNodeCmd::doIt( const MArgList &args )
+                asTexture ? MaterialXTextureNode::MATERIALX_TEXTURE_NODE_TYPEID
+                : MaterialXSurfaceNode::MATERIALX_SURFACE_NODE_TYPEID
             );
-        }
 
-        elementPath.set(materialXData->getElementPath().c_str());
-        if (elementPath.length() == 0)
-        {
-            throw mx::Exception("The element specified is not renderable.");
-        }
+            // Generate a valid Maya node name from the path string
+            const std::string nodeName = mx::createValidName(renderableElementPath);
+            _dgModifier.renameNode(node, nodeName.c_str());
 
-        // Decide what kind of node we want to create
-        bool asTexture = false;
-        if (parser.isFlagSet(kTextureFlag))
-        {
-            CHECK_MSTATUS(argData.getFlagArgument(kTextureFlag, 0, asTexture));
-        }
-        else
-        {
-            asTexture = !materialXData->elementIsAShader();
-        }
+            MFnDependencyNode depNode(node);
+            auto materialXNode = dynamic_cast<MaterialXNode*>(depNode.userNode());
+            if (!materialXNode)
+            {
+                throw mx::Exception("Unexpected DG node type.");
+            }
 
-        // Create the MaterialX node
-        MObject node = _dgModifier.createNode(
-            asTexture ? MaterialXTextureNode::MATERIALX_TEXTURE_NODE_TYPEID
-            : MaterialXSurfaceNode::MATERIALX_SURFACE_NODE_TYPEID
-        );
+            materialXNode->setData(documentFilePath, renderableElementPath.c_str(), std::move(materialXData));
 
-        // Generate a valid Maya node name from the path string
-        {
-            const std::string nodeName = mx::createValidName(elementPath.asChar());
-            _dgModifier.renameNode(node, nodeName.c_str());
+            nodeNames.append(nodeName.c_str());
         }
 
-        MFnDependencyNode depNode(node);
-        auto materialXNode = dynamic_cast<MaterialXNode*>(depNode.userNode());
-        if (!materialXNode)
+        status = _dgModifier.doIt();
+        if (!status)
         {
-            throw mx::Exception("Unexpected DG node type.");
+            unsigned int nodeNameCount = nodeNames.length();
+            std::string nodeString;
+            if (nodeNameCount)
+            {
+                nodeString = nodeNames[0].asChar();
+                for (unsigned int i = 1; i < nodeNameCount; i++)
+                {
+                    nodeString += std::string(",") + nodeNames[i].asChar();
+                }
+            }
+            throw mx::Exception(nodeString);
         }
-
-        materialXNode->setData(documentFilePath, elementPath, std::move(materialXData));
-        _dgModifier.doIt();
-
-        MString message("Created ");
-        message += materialXNode->typeName();
-        message += " node: ";
-        message += materialXNode->name();
-        MGlobal::displayInfo(message);
-        return MS::kSuccess;
+        setResult(nodeNames);        
```

Comment:
Minor mod to just return the created nodes as the result list. Can thus use the return list and do something with it if desired (e.g. select them). Can't really do anything with the info message from before.